helm-org.el

```diff
@@ -275,6 +277,36 @@ nothing to CANDIDATES."
     (helm-exit-and-execute-action
      'helm-org-insert-link-to-heading-at-marker)))
 
+(defun helm-org-refile-within-buffer (buffer)
+  "Refile current heading to another heading in BUFFER.
+Interactively, BUFFER is the current buffer."
+  (interactive (list (current-buffer)))
```

Comment:
This one is useful as a standalone command, not just as an action.  The user can refile the entry at point to any other heading within the buffer, using Helm to prompt for the target heading.  This was originally a function in my personal config, but I realized it belongs in Helm directly.  :)

Of course it also works as an action by calling it with point at a different heading, which is what we do in `helm-org-refile-within-buffer-action` using the `org-with-point-at` macro.